### DISCUSSION TITLE
PLEASE VOTE - Poll for new features

### DISCUSSION THREAD

**Discussion by BrightDV** [BrightDV/BoxBox#52] [Polls]
Hello all!
I have some ideas for the development of the app, but I want you to choose what to do first and if you want it or not.

My first idea is to add a comment section under each article from Formula 1, and maybe some likes too. It would use [Supabase](https://supabase.com/) which is an open-source alternative to Google's Firebase (in order to manage databases).
My second thought is about making a personalized feed: you follow some tags, like some drivers, teams… and you get the news related to them.
If you have any idea of a possible feature, let me know!
Also, I will try to add a live timing screen with much more data than the current one, but the tests need to be during a session…

Thank you all, and happy new year! :)

**Comment by sinfullad** [comment]
I don't read articles very often and the F1 subreddit is a good enough news aggregator (not for engaging in the comments though)

But I would personally prefer better schedule system like adding live countdown with optional reminder notifications like when it is RAWE CEEK and/or a just few hours before each sessions.
They would be a good addition to add on top of or replace the current LIVE header

**Reply by BrightDV** [reply]
Thanks for your feedback. I have added some other sources and subscription-like feed.
I have created a better indicator for header and I am currently working on the notification.
![screenshot](https://user-images.githubusercontent.com/92821484/218308810-daea7c0d-ffda-4c94-8276-341da164f1ea.png)
It stills need some improvements but it is better and clearer than the older one.

**Comment by Joshndroid** [comment]
I read some articles that interest me but not everything.

I would be happy with a couple tabs possibly one for team and one for driver. I have a team I have followed for years but also have a driver or 2 that I like regardless of the team they drive for.

Some more stat's and info would be nice during a race weekend. I don't know what that would look like but I would enjoy some more.

I typically have to watch the race as a replay due to timezone so would like any race/session notification to be a toggle.

Great open source app

**Reply by BrightDV** [reply]
I may do that, but the stats are relatively limited as I parse [the official website](https://f1.com). However, I can add a screen for teams' stats, like the one for the driver.
Furthermore, there is now a “personalized” feed, you subscribe to some topics you like (team/driver…) and you will see only articles related to them.

**Reply by Joshndroid** [reply]
Thanks mate, great app!

**Comment by arminolat** [comment]
Having reminders in form of a notification before the FP/quali/race starts would be amazing! The app is my go to source as it is, but I need another app just for the notifications at this time.

**Reply by BrightDV** [reply]
Thanks for the suggestion!
Sadly, the notifications are the thing the most buggy in the app, as I still do not know why the notifications are not showing, but I will dig in it!
I have to add it, because it is a much-needed feature.
Also, there is an issue opened for it, I will track the progress there in the future: #9.

**Comment by sk8tz** [comment]
Hi, kudos, this app is becoming my goto for following F1. Some features suggestions. 
Schedule for the year.
Count down to next race weekend. 
Time of fp1 etc based on local timezone. 
Weather for the track weekend.
Which tires for the track.
I like the live timing, you have the fastest sector times while live, what I like to see is from the fastest time possible from the drivers fastest sectors.

Thanking you in advance for implementation of any the suggestions

**Reply by BrightDV** [reply]
Thanks for the suggestions! Some of them are already in the app.
> Schedule for the year.

There is already a schedule of the *current* year. It is the icon in the bottom right corner.
> Count down to next race weekend.

Go to Schedule > Tap the next weekend > View results. There, you will see the countdowns to Free Practices, Qualifying, (Sprint), and Race.
> Time of fp1 etc based on local timezone.

The countdowns are based on local timezone.
> Weather for the track weekend.

I thought about it, but I do not know where to add it. Maybe an icon, but forecast for which hour? Or one prevision for each session? What do you think? In the meantime (I know that it is not very convenient), there is always an article before each GP with the weather forecast: [forecast for the Miami GP.](https://www.formula1.com/en/latest/article.whats-the-weather-forecast-for-the-2023-miami-grand-prix.3lHl10kBrrTYTHlu6I0Q7Q.html)
> Which tires for the track.

I do not know how to retrieve the data, but there is [also an article from the Formula 1 before each GP, here the Azerbaijan GP.](https://www.formula1.com/en/latest/article.2023-azerbaijan-grand-prix-tyres-pirelli-teams-drivers.1xGQAuax0HLdQXf0DzSw9H.html) Sometimes there is also an article with the different tyre strategies and the tyres available to each driver before the race.
> I like the live timing, you have the fastest sector times while live, what I like to see is from the fastest time possible from the drivers fastest sectors.

If you mean the screen during a session, it is only a web view of the official website, so I can't change it. I tried to recreate it myself, but there were issues that could not be fixed.

**Comment by RekZix** [comment]
Being able to switch between the time and/or the countdown would be cool as i dont want to see a countdown but the actual time so i can schedule properly without having to calculate or lookup the actual start time of quali or race.

**Reply by BrightDV** [reply]
Do you think it would be better to have a setting in the parameters or a switch directly under the countdown/hour?
Something like
![idea](https://github.com/BrightDV/BoxBox/assets/92821484/18cae0c0-c828-400a-a7ed-47bef01fc346)

I think the second solution is the best because otherwise there would be too many options (which you set once) in the settings.

**Reply by RekZix** [reply]
Yes i think the second Option is the easiest for everyone to choose which display is preferred. It would be cool if it remembers the last position but i would only put that in for the reason if people prefer the Countdown.

**Reply by BrightDV** [reply]
So this is the final result: [Screencast of Box, Box!.](https://github-production-user-asset-6210df.s3.amazonaws.com/92821484/244433078-dd2e5cd3-8318-4dcf-aff1-04e3b4986cc6.webm)
The setting is saved locally, so you will not have to change it all the time.

Also, I searched a little about calendars with Flutter, adding an event to the calendar should not be that much difficult. So, I may add a button to add a session directly to the calendar through the calendar app (ref: [add_2_calendar](https://pub.dev/packages/add_2_calendar)).

**Reply by RekZix** [reply]
Wow. That looks good and the calendar feature sounds cool.

**Reply by BrightDV** [reply]
This is now available in the latest release! Thanks for the suggestion.

**Reply by RekZix** [reply]
Great. Is there any way to donate to you :P ?

**Reply by BrightDV** [reply]
> Great. Is there any way to donate to you :P ?

Wow. That means a lot! Thanks!
However, I don't have anything setup now. I may add something in the future, but for now you can save your money ;)
